src/main/java/org/broadinstitute/hellbender/tools/exome/orientationbiasvariantfilter/PreAdapterOrientationScorer.java

```diff
@@ -0,0 +1,80 @@
+package org.broadinstitute.hellbender.tools.exome.orientationbiasvariantfilter;
+
+import com.google.cloud.dataflow.sdk.repackaged.com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.math3.linear.Array2DRowRealMatrix;
+import org.apache.commons.math3.linear.RealMatrix;
+import org.broadinstitute.hellbender.tools.picard.analysis.artifacts.SequencingArtifactMetrics;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class handles the reading of a picard metrics file PreAdapterDetailMetrics.
+ */
+public class PreAdapterOrientationScorer {
+
+    // rows {PRO, CON} x cols {ref, alt}
+    final static int PRO = 0;
+    final static int CON = 1;
+    final static int REF = 0;
+    final static int ALT = 1;
+
+    /**
+     * Gets PreAdapterQ collapsed over contexts.
+     *
+     * rows {PRO, CON} x cols {ref, alt}
+     * @param metrics metrics usuall read from a picard preadapter detail file.
+     * @return mapping to score all orientation bias artifact modes to a PreAdapterQ score.  This score can be used as a bam-file level score for level of artifacts.
+     */
+    @VisibleForTesting
+    static Map<ArtifactMode, RealMatrix> countOrientationBiasMetricsOverContext(final List<SequencingArtifactMetrics.PreAdapterDetailMetrics> metrics) {
+        Utils.nonNull(metrics, "Input metrics cannot be null");
+
+        // Artifact mode to a matrix
+        final Map<ArtifactMode, RealMatrix> result = new HashMap<>();
+
+        // Collapse over context
+        for (SequencingArtifactMetrics.PreAdapterDetailMetrics metric : metrics) {
+            final ArtifactMode key = ArtifactMode.of(metric.REF_BASE, metric.ALT_BASE);
+            result.putIfAbsent(key, new Array2DRowRealMatrix(2, 2));
```

Comment:
That is a bit more difficult, unless I am missing something.  Since I am collapsing over contexts and I do not know the order of entries in the metrics file.

No action.